src/tendril.rs

```diff
@@ -906,7 +1062,11 @@ impl<F> Tendril<F>
     }
 }
 
-impl DerefMut for Tendril<fmt::Bytes> {
+// There's no need to worry about locking on an atomic Tendril, because it makes it unique as
+// soon as you do that.
```

Comment:
Does it? As far as I can tell, regardless of atomicity (so you didn’t introduce this issue, it existed before), this `DerefMut` impl introduces mutable aliasing for shared tendrils. I think it should call `make_owned_with_capacity` (or perhaps a new `make_owned` method that doesn’t do the growing part) before `assume_buf`.